From: Carsten Schoenert <c.schoenert@t-online.de>
Date: Wed, 21 Jun 2017 18:41:53 +0200
Subject: FTBFS s390x: Use jit/none/AtomicOperations-sparc.h on s390x

s390x needs also to use the atomic operation instructions that are also
used on sparc as they are generic enough.
---
 mozilla/js/src/jit/AtomicOperations.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mozilla/js/src/jit/AtomicOperations.h b/mozilla/js/src/jit/AtomicOperations.h
index a8970b0..1cf22bf 100644
--- a/mozilla/js/src/jit/AtomicOperations.h
+++ b/mozilla/js/src/jit/AtomicOperations.h
@@ -364,6 +364,9 @@ AtomicOperations::isLockfreeJS(int32_t size)
 #elif defined(__aarch64__)
 # if defined(__clang__) || defined(__GNUC__)
 #  include "jit/arm64/AtomicOperations-arm64.h"
+# elif defined(__s390__)
+  // The sparc file is more or less generic and works on s390x.
+#  include "jit/none/AtomicOperations-sparc.h"
 # else
 #  error "No AtomicOperations support for this platform+compiler combination"
 # endif
